src/components/Api.js

```diff
@@ -1,5 +1,125 @@
 export default class Api {
   constructor(options){
-    
+    this._url = options.baseUrl;
+    this._header = options.headers;
+  }
+
+  getCards(){ // working
+    return fetch(`${this._url}/cards`, {
+      method: "GET",
+      headers: this._header
+    })
+    .then(res => res.ok ? res.json() : Promise.reject(`Error: ${res.status}`))
+    .then(data => {
+      return data
+    }).catch(err => {
+      console.error(err)
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.